code/game/gamemodes/miniantags/pulsedemon/pulsedemon.dm

```diff
@@ -0,0 +1,823 @@
+		LAZYADD(apc_images[apc_turf], apc_image)
+		client.images += apc_image
+
+/mob/living/simple_animal/demon/pulse_demon/emp_act(severity)
+	if(emp_debounce)
+		return
+
+	. = ..()
+	visible_message("<span class ='danger'>[src] [pick("fizzles", "wails", "flails")] in anguish!</span>")
+	playsound(get_turf(src), pick(hurt_sounds), 30, TRUE)
+	throw_alert(ALERT_CATEGORY_NOREGEN, /obj/screen/alert/pulse_noregen)
+	switch(severity)
+		if(EMP_LIGHT)
+			adjustHealth(round(max(initial(health) / 4, round(maxHealth / 8))))
+			regen_lock = 3
+		if(EMP_HEAVY)
+			adjustHealth(round(max(initial(health) / 3, round(maxHealth / 6))))
+			regen_lock = 5
+	emp_debounce = TRUE
+	addtimer(VARSET_CALLBACK(src, emp_debounce, FALSE), 0.1 SECONDS, TIMER_UNIQUE | TIMER_OVERRIDE)
+
+/mob/living/simple_animal/demon/pulse_demon/proc/try_attack_mob(mob/living/L)
+	if(!is_under_tile() && L != src)
+		do_attack_animation(L)
+		try_shock_mob(L)
+
+/mob/living/simple_animal/demon/pulse_demon/UnarmedAttack(atom/A)
+	if(isliving(A))
+		var/mob/living/L = A
+		try_attack_mob(L)
+	else if(isitem(A) && !is_under_tile())
+		var/obj/item/O = A
+		var/obj/item/stock_parts/cell/C = O.get_cell()
+		if(C && C.charge)
+			C.use(min(C.charge, power_drain_rate))
+			adjust_charge(min(C.charge, power_drain_rate))
+			visible_message("<span class='notice'>[src] touches the [O] and drains its power!</span>", "<span class='notice'>You touch the [O] and drain its power!</span>")
+
+/mob/living/simple_animal/demon/pulse_demon/attack_hand(mob/living/carbon/human/M)
+	if(is_under_tile())
+		to_chat(M, "<span class='danger'>You can't interact with something that's under the floor!</span>")
+		return
+	switch (M.intent)
+		if(INTENT_HELP)
+			visible_message("<span class='notice'>[M] [response_help] [src].</span>")
+		if(INTENT_DISARM, INTENT_GRAB)
+			visible_message("<span class='notice'>[M] [response_disarm] [src].</span>")
+		if(INTENT_HELP)
+			visible_message("<span class='warning'>[M] [response_harm] [src].</span>")
+	try_attack_mob(M)
+
+/mob/living/simple_animal/demon/pulse_demon/attackby(obj/item/O, mob/living/user)
+	if(is_under_tile())
+		to_chat(user, "<span class='danger'>You can't interact with something that's under the floor!</span>")
+		return
+	var/obj/item/stock_parts/cell/C = O.get_cell()
+	if(C && C.charge)
+		C.use(min(C.charge, power_drain_rate))
+		adjust_charge(min(C.charge, power_drain_rate))
+		to_chat(user, "<span class='warning'>You touch [src] with [O] and [src] drains it!</span>")
+		to_chat(src, "<span class='notice'>[user] touches you with [O] and you drain its power!</span>")
+	visible_message("<span class='notice'>[O] goes right through [src].</span>")
+	try_shock_mob(user, O.siemens_coefficient)
+
+/mob/living/simple_animal/demon/pulse_demon/ex_act()
+	return
+
+/mob/living/simple_animal/demon/pulse_demon/CanPass(atom/movable/mover, turf/target, height)
+	. = ..()
+	// maybe a prob on this? prob only for weak ions? discover in testing
```

Comment:
Can you take a look at this, or remove the comment if it checks out? Also might be worth adding a taser interaction here, though that'd probably be something to check out later.